examples/recreate_vm.py

```diff
@@ -0,0 +1,149 @@
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+"""
+Show how to recreate VM while maintaining VM permissions
+
+vm_name - VM to recreate
+"""
+
+import logging
+import time
+
+import ovirtsdk4 as sdk
+import ovirtsdk4.types as types
+import sys
+
+
+vm_name = 'vm'
+
+# Create the connection to the server:
+
+connection = sdk.Connection(
+    url='https://localhost.localdomain:8443/ovirt-engine/api',
+    username='admin@internal',
+    password='redhat123',
+    ca_file='Environments/ca.cer',
+    debug=True,
+    log=logging.getLogger(),
+)
+
+system_service = connection.system_service()
+
+vms_service = system_service.vms_service()
+
+vm = vms_service.list(search='name=%s' % vm_name)
+if not vm:
+    print('VM doesnt exist')
+    sys.exit(0)
+
+vm = vm[0]
+vm_template = connection.follow_link(vm.template)
+template_id = vm_template.id
+template_name = vm_template.name
+template_base_template = vm_template.version.base_template
+
+if vm.use_latest_template_version == True:
+    print('VM should not use latest template version')
+    sys.exit(0)
+
+if vm.status != types.VmStatus.DOWN:
+    print('VM status should be down')
+    sys.exit(0)
+
+
+vm_service = vms_service.vm_service(vm.id)
+templates_service = system_service.templates_service()
+
+# Remove the disks from the VM
+
+disk_attachments_service = vm_service.disk_attachments_service()
+disks_service = system_service.disks_service()
+
+for disk_attachment in disk_attachments_service.list():
+    disk_attachment_service = disk_attachments_service.attachment_service(disk_attachment.id)
+    disk_service = disks_service.disk_service(disk_attachment.disk.id)
+    disk_service.remove()
+
+
+# Create new template from the VM
+
+new_template = templates_service.add(
+    template=types.Template(
+        version=types.TemplateVersion(
+            base_template=template_base_template,
+            version_name='99',
+            version_number=99
+        ),
+        name=template_name,
+        vm=types.Vm(
+            id=vm.id
+        )
+    )
+)
+new_template_service = templates_service.template_service(new_template.id)
+
+# Wait till the status of the template is OK, as that means that it is
+# completely created and ready to use
+
+while True:
+    time.sleep(5)
+    new_template = new_template_service.get()
+    if new_template.status == types.TemplateStatus.OK:
+        break
+
+# Change the VM template to the new version
+
+try:
+    vm_service.update(
+        types.Vm(
+            template = types.Template(
+                id=new_template.id
+            )
+        )
+    )
+except:
+    pass
```

Comment:
why do we have the try-except block here and further down?

Reply:
because the call is failing - it can't query the vm and return it because the vm is already deleted and recreated

If i don't put try catch the script will exit